src/components/EmbeddedPDF.tsx

```diff
@@ -26,15 +28,25 @@ const EmbeddedPDF = ({ document, documentPassageMatches = [], passageIndex = nul
   // Ensure the instance of the PDF client is not reset on re-render
   // otherwise we lose the ability to interact with the pdf
   // eslint-disable-next-line react-hooks/exhaustive-deps
-  const { changePage, addAnnotations } = useMemo(() => pdfPreview, [document, adobeKey]);
+  const { changePage, registerPassages } = useMemo(() => pdfPreview, [document, adobeKey]);
 
   useEffect(() => {
     changePage(passageIndex, documentPassageMatches);
   }, [changePage, passageIndex, documentPassageMatches]);
 
   useEffect(() => {
-    addAnnotations(documentPassageMatches, startingPassageIndex);
-  }, [addAnnotations, documentPassageMatches, startingPassageIndex]);
+    setIsLoading(true);
+    registerPassages(documentPassageMatches, startingPassageIndex).finally(() => {
+      setIsLoading(false);
+    });
+  }, [registerPassages, documentPassageMatches, startingPassageIndex]);
+
+  useEffect(() => {
+    if (searchStatus === "loading" || isLoading) {
+      return setIsLoading(true);
```

Comment:
Question: Does setting `isLoading` to true when it is already true cause a re-render?